File: src/modules/presentation/io/tripled/social/client/presentation/controller/FollowController.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.FollowUserUseCase;
import io.tripled.social.client.presentation.Controller;
import io.tripled.social.client.presentation.Output;
import io.tripled.social.client.presentation.request.FollowCliRequest;

public class FollowController implements Controller<FollowCliRequest> {

  private FollowUserUseCase followUserUseCase;

  public FollowController(FollowUserUseCase followUserUseCase) {
    this.followUserUseCase = followUserUseCase;
  }

  public void execute(FollowCliRequest request, Output output) {
    output.print(followUserUseCase.follow(request.getUserName(), request.getUserNameToFollow()));
  }


}

File: src/main/kotlin/io/tripled/social/client/infrastructure/cli/controller/FollowController.kt
package io.tripled.social.client.infrastructure.cli.controller

import io.tripled.social.client.application.FollowUser
import io.tripled.social.client.infrastructure.cli.Controller
import io.tripled.social.client.infrastructure.cli.Output
import io.tripled.social.client.infrastructure.cli.request.FollowCliRequest

class FollowController(private val followUser: FollowUser) : Controller<FollowCliRequest> {
    override fun execute(request: FollowCliRequest, output: Output) {
        output.print(followUser.follow(request.userName, request.userNameToFollow))
    }
}
